packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.types.ts

```diff
@@ -102,6 +104,7 @@ export type UseTreeViewItemsSignature = TreeViewPluginSignature<{
   events: UseTreeViewItemsEventLookup;
   state: UseTreeViewItemsState<any>;
   contextValue: UseTreeViewItemsContextValue;
+  experimentalFeatures: 'indentationAtItemLevel';
```

Comment:
I'm just checking: is this correct? 🤔 
I would assume that we'd want something like:
```ts
type UseTreeViewItemsExperimentalFeatures {
  indentationAtItemLevel: boolean;
}
```

Reply:
I took for granted that the shape of `experimentalFeatures` was always an object with the feature name as the key and a boolean as the value.

So I kept the definition at the plugin level as simple as possible with just the keys.
And then I do:


```ts
export type TreeViewExperimentalFeatures<
  TSignatures extends readonly TreeViewAnyPluginSignature[],
> = { [key in MergePluginsProperty<TSignatures, 'experimentalFeatures'>]?: boolean };
```